airflow/providers/google/cloud/transfers/bigquery_to_gcs.py

```diff
@@ -139,14 +139,34 @@ def execute(self, context):
             location=self.location,
             impersonation_chain=self.impersonation_chain,
         )
-        conn = hook.get_conn()
-        cursor = conn.cursor()
-        cursor.run_extract(
-            source_project_dataset_table=self.source_project_dataset_table,
-            destination_cloud_storage_uris=self.destination_cloud_storage_uris,
-            compression=self.compression,
-            export_format=self.export_format,
-            field_delimiter=self.field_delimiter,
-            print_header=self.print_header,
-            labels=self.labels,
+
+        source_project, source_dataset, source_table = split_tablename(
+            table_input=self.source_project_dataset_table,
+            default_project_id=hook.project_id,
+            var_name='source_project_dataset_table',
         )
+
+        configuration = {
+            'extract': {
+                'sourceTable': {
+                    'projectId': source_project,
+                    'datasetId': source_dataset,
+                    'tableId': source_table,
+                },
+                'compression': self.compression,
+                'destinationUris': self.destination_cloud_storage_uris,
+                'destinationFormat': self.export_format,
+            }
+        }  # type: Dict[str, Any]
```

Comment:
```suggestion
        configuration: Dict[str, Any] = {
            'extract': {
                'sourceTable': {
                    'projectId': source_project,
                    'datasetId': source_dataset,
                    'tableId': source_table,
                },
                'compression': self.compression,
                'destinationUris': self.destination_cloud_storage_uris,
                'destinationFormat': self.export_format,
            }
        }
```
We can use type hints like this as the master branch doesn't have to support Py 2.7